### PR TITLE
Issue/1522 bootstrap service crash part2

### PR DESCRIPTION
**Pull request by valllllll2000**
#### The solution
Extracted classes to make the code more testable + added tests
this is part 2 of https://github.com/akvo/akvo-flow-mobile/pull/1715
#### Reviewer Checklist
* [ ] Added an explanation about the work done
* [ ] Connected the PR and the issue on Zenhub
* [ ] Added a test plan to the issue
* [ ] Updated the copyright header
* [ ] Formatted the code per our style guide
* [ ] Added a documentation (if relevant)

### CODE REVIEW

File: app/src/test/java/org/akvo/flow/service/bootstrap/SurveyMapperTest.kt [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    fun getSurveyIdFromFilePathShouldReturnEmptyIfFolderMissing() {
+        val surveyId = surveyMapper.getSurveyIdFromFilePath("file.xml")
+        assertEquals("", surveyId)
+    }
+
+    //TODO: separate into several methods
+    @Test
+    fun getSurveyIdFromFilePathShouldReturnCorrectIdFromFolder() {
+        var surveyId = surveyMapper.getSurveyIdFromFilePath("123/file.xml")
+        assertEquals("123", surveyId)
```
Comment (tangrammer) [inline app/src/test/java/org/akvo/flow/service/bootstrap/SurveyMapperTest.kt:76]: I expected to get 1234 here in relation to https://github.com/akvo/akvo-flow-mobile/pull/1716/files#diff-712dd749f6a46fcacbf07e48bd9d8f403c91161441802a27bca2277070a7b8bfR54

**Review by tangrammer — COMMENTED** [review]
just left a minor question related to tests

File: app/src/test/java/org/akvo/flow/service/bootstrap/SurveyMapperTest.kt (hunk 1 above)
Comment (valllllll2000) [reply app/src/test/java/org/akvo/flow/service/bootstrap/SurveyMapperTest.kt:76]: I don't understand how one can get 1234, it parses the name and extracts the survey id which is part of the path

Comment (valllllll2000) [reply app/src/test/java/org/akvo/flow/service/bootstrap/SurveyMapperTest.kt:76]: no here we don't use the mocked one, we use the file path provided, ie "123/form.xml" means the surveyId is 123

**Review by tangrammer — APPROVED** [review]

**Pull request merged**